src/filepattern/cpp/include/filepattern.h

```diff
@@ -32,12 +39,13 @@ using Tuple = std::tuple<Map, std::vector<std::string>>;
 using Tuple = std::tuple<Map, std::vector<fs::path>>;
 #endif
 
+
 class PatternObject; // forward declaration
 class FILEPATTERN_EXPORT FilePattern {
 
     public:
 
-        FilePattern(const std::string& path, const std::string& filePattern="", const std::string& block_size="", bool recursive=false, bool suppressWarnings=false);
+        FilePattern(const std::string& path, const std::string& filePattern="", const std::string& block_size="", bool recursive=false, bool suppressWarnings=false, const std::vector<std::string>& file_array={});
```

Comment:
Instead of this, have two constructors. One as the old one and a new one with the newly added argument.